lib/Dialect/TritonGPU/Transforms/OptimizeDotOperands.cpp

```diff
@@ -259,30 +259,25 @@ class HoistLayoutConversion : public OpRewritePattern<ConvertLayoutOp> {
     SetVector<Operation *> slice;
     BackwardSliceOptions opt;
     opt.omitBlockArguments = true;
-    // TODO(jlebar): Is this filter redundant with omitBlockArguments == true?
-    // That is, is it possible to get into a different region without going
-    // through a block argument?
-    opt.filter = [&](Operation *op) {
```

Comment:
I'm not an expert here, but what are the scenarios where we can jump outside the block without going through the arguments besides a constant? Just trying to verify the impact of this part. Maybe @chsigg can give us some MLIR insights as well?

Reply:
In theory, it could be anything, including a load.  The important part is that it's not affected by the block and not changing over iterations and can therefore be propagated just as if it were inside the block.

Reply:
That makes sense. I guess Justin was correct about this being sort of redundant.